src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetFeatureInfoTest.java

```diff
@@ -872,4 +876,46 @@ public void testScaleTolerance() throws Exception {
        assertPixel(image, 150, 150, Color.BLUE);
    }
    
+   /**
+    * Test GetFeatureInfo on a group layer with some no-queryable layers
+    * @throws Exception
+    */
+   @Test
+   public void testGroupLayerWithNotQueryableLayers() throws Exception {
+       Catalog catalog = getCatalog();
+       CatalogFactory factory = catalog.getFactory();
+       WorkspaceInfo workspace = catalog.getWorkspaceByName(MockData.CITE_PREFIX);
+       String groupLayer = "glqueryable";
+       
+       LayerInfo buildingsLayer = ModificationProxy.unwrap(catalog.getLayerByName(getLayerId(MockData.BUILDINGS)));
+       LayerInfo bridgesLayer = ModificationProxy.unwrap(catalog.getLayerByName(getLayerId(MockData.BRIDGES)));
+       LayerInfo forestLayer = ModificationProxy.unwrap(catalog.getLayerByName(getLayerId(MockData.FORESTS)));
+       
+       // Only last layer will be queryable.
+       buildingsLayer.setQueryable(false);
+       bridgesLayer.setQueryable(false);
+       forestLayer.setQueryable(true);
```

Comment:
Oh no, you unwrapped because you did not want to save the layers after changing them? This is a severe mistake, even for test code. Remove the unwrapping and save them after changing the queryable flag.